snapcraft/internal/build_providers/_base_provider.py

```diff
@@ -78,33 +139,33 @@ def _umount(self, *, mountpoint: str) -> None:
     def _mount_snaps_directory(self) -> None:
         """Mount the host directory with snaps into the provider."""
 
-    def _unmount_snaps_directory(self) -> None:
-        self._umount(mountpoint=self._SNAPS_MOUNTPOINT)
-
-    @abc.abstractmethod
-    def _push_file(self, *, source: str, destination: str) -> None:
-        """Push a file into the instance."""
-
     @abc.abstractmethod
-    def create(self) -> None:
-        """Provider steps needed to create a fully functioning environment."""
+    def _unmount_snaps_directory(self) -> None:
+        """Mount the host directory with snaps into the provider."""
 
     @abc.abstractmethod
-    def destroy(self) -> None:
-        """Provider steps needed to ensure the instance is destroyed.
-
-        This method should be safe to call multiple times and do nothing
-        if the instance to destroy is already destroyed.
+    def mount_project(self) -> None:
+        """Provider steps needed to make the project available to the instance.
         """
 
     @abc.abstractmethod
     def provision_project(self, tarball: str) -> None:
         """Provider steps needed to copy project assests to the instance."""
 
-    @abc.abstractmethod
-    def mount_project(self) -> None:
-        """Provider steps needed to make the project available to the instance.
-        """
+    def execute_step(self, step: steps.Step) -> None:
+        self._run(command=["snapcraft", step.name])
+
+    def pack_project(self, *, output: Optional[str] = None) -> None:
+        command = ["snapcraft", "snap"]
+        if output:
+            command.extend(["--output", output])
+        self._run(command=command)
+
+    def clean_project(self) -> bool:
+        if not os.path.exists(self.provider_project_dir):
+            return False
+        shutil.rmtree(self.provider_project_dir)
+        return True
```

Comment:
Perhaps this would be more pythonic:

```python
try:
    shutil.rmtree(self.provider_project_dir)
    return True
except FileNotFoundError:
    return False
```